### PR TITLE
Test cryptsetup in FIPS mode

### PR DESCRIPTION
**Pull request by dehai**
Related test case:Case 1528909  - FIPS: cryptsetup
https://bugzilla.suse.com/tr_show_case.cgi?case_id=1528909
This case is verify the command of cryptsetup whether can be work in FIPS mode
Local openQA validation running: http://147.2.212.179/tests/330

### CODE REVIEW

File: tests/console/cryptsetup.pm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# Case 1528909  - FIPS: cryptsetup
+
+use base "consoletest";
+use strict;
+use testapi;
+
+sub run() {
+
+    my $slef        = shift;
```
Comment (okurz) [inline tests/console/cryptsetup.pm]: `$slef`? funny :-) It's a typo but you never use `$slef` so better just delete the line or replace by `my $self = @_;`

File: tests/console/cryptsetup.pm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    assert_script_run("echo -e $cryptpasswd | cryptsetup -q luksFormat /test.dm");
+    wait_still_screen;
+    assert_script_run("echo -e $cryptpasswd | cryptsetup -q luksOpen /test.dm dmtest");
+    wait_still_screen;
+
+    # format the dmtest and mount it with /test
+    assert_script_run("mkfs.ext4 /dev/mapper/dmtest");
+    wait_still_screen;
+    assert_script_run("mkdir /test;mount /dev/mapper/dmtest /test");
+    wait_still_screen;
```
Comment (okurz) [inline tests/console/cryptsetup.pm]: all the `wait_still_screen` wait for 7 seconds by default which might be a bit long. As you are calling `assert_script_run` already I would assume you don't need to wait at all. Are the `wait_still_screen` necessary or can you just remove them?

File: tests/console/cryptsetup.pm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# Case 1528909  - FIPS: cryptsetup
+
+use base "consoletest";
+use strict;
+use testapi;
+
+sub run() {
+
+    my $slef        = shift;
+    my $cryptpasswd = 'susetesting';
```
Comment (okurz) [inline tests/console/cryptsetup.pm]: would it be ok to use just the default `$testapi::password`? E.g.

```perl
my $cryptpasswd = $testapi::password;
```

File: tests/console/cryptsetup.pm [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    # format the dmtest and mount it with /test
+    assert_script_run("mkfs.ext4 /dev/mapper/dmtest");
+    wait_still_screen;
+    assert_script_run("mkdir /test;mount /dev/mapper/dmtest /test");
+    wait_still_screen;
+
+    # build some directory and files in the mount point ,try to catch file and delete directory and file
+    assert_script_run("cd /test");
+    assert_script_run("for x in `seq 100`; do mkdir d\$x; date > d\$x/f\$x.log; done");
```
Comment (okurz) [inline tests/console/cryptsetup.pm]: As you don't want perl to evaluate the variables I recommend to use single ticks instead, e.g.

```perl
assert_script_run('for x in `seq 100`; do mkdir d$x; date > d$x/f$x.log; done');
```

File: tests/console/cryptsetup.pm [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    assert_script_run("cat d39/f39.log");
+    assert_script_run("rm -rf d4*");
+
+    # release the mount point and cryptsetup luksClose dmtest
+    assert_script_run("cd ~");
+    assert_script_run("umount /test");
+    assert_script_run("cryptsetup luksClose dmtest");
```
Comment (okurz) [inline tests/console/cryptsetup.pm]: just to cleanup everything it would be nice to also delete `/test.dm`

**Comment by okurz** [comment]
please see inline comments

**Comment by dehai** [comment]
@okurz ,ok., I saw your comments, and will modify it .some non-necessary  `wait_still_screen` , I will delete them.

**Comment by dehai** [comment]
Local openQA validation running: http://147.2.212.179/tests/331

**Pull request merged**